service/lib/dinstaller/software.rb

```diff
@@ -68,10 +72,23 @@ def select_product(name)
       return if name == @product
       raise ArgumentError unless @products[name]
 
+      # TODO: if a question is asked here, there is no web UI to handle it.
+      # Is it a problem?
+      # testing_question if ENV["DINSTALLER_TEST_QUESTIONS"] == "1"
```

Comment:
testing code leftover?

Reply:
Yes, but a good opportunity to ask about the TODO:

What should happen when some code asks a Question and the web UI has no component to display it?
Currently the code deadlocks waiting for an answer that will not come.
Arguably it will only happen during development.
We could add a `displayed?` property to Questions and `raise` an error if a Question is not displayed within 20 seconds.